### DISCUSSION TITLE
2022-07 Playtest Request

### DISCUSSION THREAD

**Discussion by BenjamenMeyer** [vegastrike/Vega-Strike-Engine-Source#715] [Bugs]
@royfalk requested via our [gitter channel](https://gitter.im/vegastrike/community?at=62d84d308fe56a38b5ac5ec2):
```
Can someone please playtest the game in master vs. the game in the latest stable release or master minus two months? I'm testing a new PR and I'm having trouble deciding if the game's OK or it became worse because of something recent. As I haven't played the game (seriously) in a long time, I'm not sure what to expect.
Things I saw:
Oswald was sometimes flying funny, rotating as if it was colliding with something.
Game crash when firing on a capital ship at close range (maybe collision).
Not seeing the sort of aggression I expect after killing Oswald or other neutral ships. This may be due to the fact I hacked a rather aggressive ship and the AI doesn't want to pick a fight.
```

Things to test:
- [ ] `Oswald was sometimes flying funny, rotating as if it was colliding with something.` :arrow_right: Watch and interact with Oswald
- [x] `Game crash when firing on a capital ship at close range (maybe collision).` :arrow_right: `attack a capital ship`
- [x] `Not seeing the sort of aggression I expect after killing Oswald or other neutral ships. This may be due to the fact I hacked a rather aggressive ship and the AI doesn't want to pick a fight.` :arrow_right: ??

**Comment by stephengtuggy** [comment]
Yes, I have observed all three of these game behaviors myself while play-testing during the last few months.

Regarding point 3, the lack of aggression, I filed https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/695 (probably the same root cause). Regarding point 1, when I collide my ship with Oswald's, I will often see him go spinning away, and keep on spinning almost indefinitely. I suspect that that is related to point 3.

Regarding item 2: I have also seen this, but have yet to file a bug for it. Can someone file a bug?

**Comment by BenjamenMeyer** [comment]
PR #716 is part of this

**Comment by evertvorster** [comment]
Vegastrike does not start for me at all now. I will make a report soon.

**Reply by stephengtuggy** [reply]
@evertvorster This isn't still the case, is it? Please tell me Vega Strike is at least starting for you again.

**Reply by evertvorster** [reply]
Vega Strike from git master is starting fine. Sorry for the delay!

**Reply by stephengtuggy** [reply]
@evertvorster That's good. Thanks for reporting back.

**Comment by LifWirser** [comment]
This could be a good time to test valgrind with instructions in  pullrequest #719 . gdb can be tried as well but callgrind tool can show what code is actually called .  the valgrind suite has lots of fum stuff . https://valgrind.org/ for more

**Reply by LifWirser** [reply]
pulled from master both engine and assets

I lost patience with oswald at point where he instructs you to turn around and accel after 15 km I turn back around and started target practice. oswald will fight back the he kamakazed into me.. Speaking of Oswald or rather about Oswalds speaking some adjusting of sound levels to his replies , some are barley audible
at both fortress and atlantis  I could sell my standard capacitor but not buy ant. also there seems to be missing data as well - capacity and such
And also had a crash on launching - haven't repeated yet

**Reply by BenjamenMeyer** [reply]
valgrind is a neat tool; but does mess with performance and therefore timing. So sure, take it for a spin - just be aware of it's limitations too.

**Reply by stephengtuggy** [reply]
@LifWirser Yes, valgrind is pretty cool, from what I've seen. Has all kinds of capabilities. It does slow things down a LOT, though.

This week, I've done some experimenting with AddressSanitizer, sort of an alternative or competitor to valgrind. My hope was that it wouldn't slow things down so badly.

Well, that did not turn out to be the case. AddressSanitizer is still fun to tinker with, though. It has a lot of capabilities of its own.
